### PR TITLE
HOT FIX: update cpu map calculation for Windows to avoid incorrect total number of processors

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *Some platform may report incorrect total number of processors. Do not use this parameter directly to avoid this negative*

### Tickets:
 - *CVS-107300*

### CODE REVIEW

File: src/inference/src/os/win/win_system_conf.cpp [hunk 1]
```diff
@@ -29,8 +29,6 @@ void CPU::init_cpu(CPU& cpu) {
         return;
     }
 
-    cpu._processors = GetMaximumProcessorCount(ALL_PROCESSOR_GROUPS);
-
     parse_processor_info_win(base_ptr,
```
Comment (ilya-lavrenov) [inline src/inference/src/os/win/win_system_conf.cpp:32]: should we statically link hwloc in 2023.1 to avoid reinventing the wheel?

Comment (wangleis) [reply src/inference/src/os/win/win_system_conf.cpp:32]: I prefer the Windows API over adding additional libraries. And we always need to handle those corner cases when dependency library cannot report right information.

Let's validate the coverage of Windows API first. HWLOC is our backup plan if there are corner cases that the OS API cannot handle.

**Review by sunxiaoxia2022 — APPROVED** [review]

File: src/inference/src/os/win/win_system_conf.cpp [hunk 2]
```diff
@@ -77,65 +84,73 @@ void parse_processor_info_win(const char* base_ptr,
...
 
             } else {
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_ID] = _cores;
+                proc_info = cpu_init_line;
+                proc_info[CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
+                proc_info[CPU_MAP_SOCKET_ID] = _sockets;
+                proc_info[CPU_MAP_CORE_ID] = _cores;
+                if ((_processors > group_start) && (_processors <= group_end)) {
```
Comment (riverlijunjie) [inline src/inference/src/os/win/win_system_conf.cpp:132]: Could you confirm _processors belongs to (group_start, group_end] or [group_start, group_end)?

File: src/inference/src/os/win/win_system_conf.cpp [hunk 3]
```diff
@@ -77,65 +84,73 @@ void parse_processor_info_win(const char* base_ptr,
...
-            proc_count += list_len;
+            _processors += list_len;
             _cores++;
 
         } else if ((info->Relationship == RelationCache) && (info->Cache.Level == 2)) {
             MaskToList(info->Cache.GroupMask.Mask);
 
             if (4 == list_len) {
-                for (int m = 0; m < list_len; m++) {
+                if (_processors < list[list_len - 1] + base_proc) {
```
Comment (riverlijunjie) [inline src/inference/src/os/win/win_system_conf.cpp:147]: It is '<' or '<='?

Comment (wangleis) [reply src/inference/src/os/win/win_system_conf.cpp:147]: it is `<`. `=` part has been covered in following for loop.

File: src/inference/src/os/win/win_system_conf.cpp (hunk 2 above)
Comment (wangleis) [reply src/inference/src/os/win/win_system_conf.cpp:132]: (group_start, group_end]. Processor of group_start has been handled before create group_start and group_end.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**